### PR TITLE
ENH: plot_roi with contours

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Discussed IRL with @GaelVaroquaux

### CODE REVIEW

**Comment by KamalakerDadi** [comment]
Still-to-do. More tests:

- testing _plot_roi_contours
- testing whether raises an error with wrong view_type

File: nilearn/plotting/img_plotting.py [hunk 1]
```diff
@@ -603,12 +603,53 @@ def plot_epi(epi_img=None, cut_coords=None, output_file=None,
...
+    labels = np.unique(roi_data)
+    cmap = plt.cm.get_cmap(cmap)
+    color_list = cmap(np.linspace(0, 1, len(labels)))
+    for idx, label in enumerate(labels):
+        if label == 0:
+            continue
+        data = (roi_data == label)
+        data = data.astype(np.int)
+        img = new_img_like(roi_img, data, affine=roi_img.affine)
+        display.add_contours(img, levels=[0.5], colors=[color_list[idx - 1]],
```
Comment (KamalakerDadi) [inline nilearn/plotting/img_plotting.py]: What would be the good default ```levels``` threshold for labels based ROIs ?

**Comment by GaelVaroquaux** [comment]
Can you rebase / merge on master, so that circle runs.

File: nilearn/plotting/img_plotting.py (hunk 1 above)
Comment (GaelVaroquaux) [reply nilearn/plotting/img_plotting.py]: I think that the one that you used (.5) is good.

File: nilearn/plotting/tests/test_img_plotting.py [hunk 2]
```diff
@@ -52,6 +53,32 @@ def demo_plot_roi(**kwargs):
...
+def test_plot_roi_contours():
+    display = plot_roi(None)
+    data = np.zeros((91, 109, 91))
+    x, y, z = -52, 10, 22
+    x_map, y_map, z_map = coord_transform(x, y, z,
+                                          np.linalg.inv(mni_affine))
+    data[int(x_map) - 5:int(x_map) + 5, int(y_map) - 3:int(y_map) + 3,
+         int(z_map) - 10:int(z_map) + 10] = 1
+    img = nibabel.Nifti1Image(data, mni_affine)
+    _plot_roi_contours(display, img, 'RdBu', 0.1, 2.)
```
Comment (GaelVaroquaux) [inline nilearn/plotting/tests/test_img_plotting.py]: I think that I would prefer it if you used the public-facing function in the test.

**Comment by GaelVaroquaux** [comment]
Also: this is cool! Can you add it to the whats_new.

**Comment by GaelVaroquaux** [comment]
Can you merge with master, so that CI runs without failing. Thanks!

File: nilearn/plotting/tests/test_img_plotting.py (hunk 2 above)
Comment (jeromedockes) [reply nilearn/plotting/tests/test_img_plotting.py]: @KamalakerDadi I think this is the last comment to address before this is merged?

> I think that I would prefer it if you used the public-facing function in the test.

**Comment by thomasbazeille** [comment]
hello @KamalakerDadi ! Everything seemed alright for this PR, is there any particular reason why it was not merged ?

**Comment by thomasbazeille** [comment]
hello @KamalakerDadi ! Apart for a few conflicts, everything seemed alright for this PR. Can it be finished and merged to be included in the upcoming release ?

**Comment by KamalakerDadi** [comment]
>hello @KamalakerDadi ! Apart for a few conflicts, everything seemed alright for this PR. Can it be finished and merged to be included in the upcoming release ?

It's doable. Let me get back to this very soon.

**Comment by KamalakerDadi** [comment]
@thomasbazeille please let me know if anything needs to be done.

**Comment by thomasbazeille** [comment]
I didn't see it added in the what's new; apart from that everything seems there after a quick review. Will merge in a few days if I don't here from anyone.

Thanks a lot for completing this great PR Kamalakar!

**Review by thomasbazeille — APPROVED** [review]

**Review by bthirion — APPROVED** [review]
Indeed a whatsnew entry would be ideal.

**Pull request merged**